core/validation/specific_validations/towns_fund_round_four.py

```diff
@@ -95,6 +101,42 @@ def validate_project_admin_gis_provided(
         ]
 
 
+def validate_sign_off(workbook):
+    sheet = workbook.get("8 - Review & Sign-Off")
+    sheet.replace(r"", np.nan, inplace=True)
+
+    section_151_text_cells = [6, 7, 8, 9]
+    town_board_chair_cells = [13, 14, 15, 16]
+
+    fields_and_sections = []
+
+    for y_axis in section_151_text_cells:
+        if pd.isnull(sheet.iloc[y_axis, 2]):
+            fields_and_sections.append((sheet.iloc[y_axis, 1], "Section 151 Officer / Chief Finance Officer"))
+
+    for y_axis in town_board_chair_cells:
+        if pd.isnull(sheet.iloc[y_axis, 2]):
+            fields_and_sections.append((sheet.iloc[y_axis, 1], "Town Board Chair"))
+
+    section_to_approver = {
+        "Section 151 Officer / Chief Finance Officer": "an S151 Officer or Chief Finance Officer",
+        "Town Board Chair": "a programme SRO",
+    }
+
+    if len(fields_and_sections) > 0:
+        return [
+            TownsFundRoundFourValidationFailure(
+                tab="Review & Sign-Off",
+                section=section,
+                message=(
+                    f"You must fill out the {field} for this section. "
+                    f"You need to get sign off from {str(section_to_approver[section])}"
+                ),
+            )
+            for field, section in fields_and_sections
+        ]
```

Comment:
Instead of collecting fields and sections and then iterating over them, can we instantiate the Failures and add them to a list on lines 115 and 119

Reply:
```
failures = []
```

and then appending each failure as and when, returning the list if not empty?

Reply:
yeah, can even just return `failures` regardless of whether its empty or not